### PR TITLE
Extracted seo-tab to reuse it in other bundles

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR extracts the logic and template of the content seo-tab to reuse it in another bundle.

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js [hunk 1]
```diff
@@ -943,6 +948,11 @@ define([
 
                     tabs: {
                         url: navigationUrl,
+                        options: {
+                            data: function() {
+                                return this.sandbox.util.extend(false, {}, this.content.toJSON());
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js]: Why is the first parameter false? This way it is not a deep copy, is that what you want? And if no, why not use the `deepCopy` util function instead?

Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js]: tbh i have copied this line from the contact. but i will change it to `deepCopy`.

**Comment by wachterjohannes** [comment]
@danrot fixed

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/main.js [hunk 2]
```diff
@@ -37,6 +39,8 @@ define([
 
             'use strict';
 
+            SeoTab.initialize(app);
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/main.js]: Isn't `app.use` the way to register extensions? I'd say extensions are also not supposed to be loaded via requirejs.

Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/main.js]: `app.use` can only be called before start is done. this is the only way to not include the code in this extension file.

Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/main.js]: As far as I can see [here](https://github.com/sulu/sulu/blob/c1809cddf2264af2c6ac48e1d539cf1a3507b46b/src/Sulu/Bundle/AdminBundle/Resources/public/js/main.js#L131) this file is called before `app.start` is called.

Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/main.js]: no this function will be called within the `start` function. this is the way we do it always (see https://github.com/sulu/sulu/blob/develop/src/Sulu/Bundle/ContactBundle/Resources/public/js/main.js#L61)

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/extensions/seo-tab.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            },
+
+            listenForChange: function() {
+                this.sandbox.dom.on(formId, 'keyup change', function() {
+                    this.setHeaderBar();
+                }.bind(this), '.trigger-save-button');
+            },
+
+            setHeaderBar: function() {
+                this.sandbox.emit('sulu.tab.dirty');
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/extensions/seo-tab.js]: Does a default implementation make sense here? Think that could lead to confusing situations. Same for `tabInitialize`...

Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/extensions/seo-tab.js]: Wouldn't it be easier for the implementor if these methods throw an error, same as the others, which are supposed to overridden? Or are there cases where these methods won't be overridden?

Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/extensions/seo-tab.js]: Or it should at least be documented somewhere 😉

Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/extensions/seo-tab.js]: yes in other usages (if the new component structure will be used) this method don`t have to be overwritten. i will add documentation in this file.

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**